src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs

```diff
@@ -98,7 +99,7 @@ private async Task AcquireAuthorizationAsync(CancellationToken cancellationToken
             AuthenticationRequestParameters.RedirectUri = _webUi.UpdateRedirectUri(AuthenticationRequestParameters.RedirectUri);
             var authorizationUri = CreateAuthorizationUri(true);
 
-            var uiEvent = new UiEvent(AuthenticationRequestParameters.RequestContext.TelemetryCorrelationId);
+            var uiEvent = new UiEvent(AuthenticationRequestParameters.RequestContext.CorrelationId.AsMatsCorrelationId());
```

Comment:
>AsMatsCorrelationId() [](start = 99, length = 21)

why is this needed?

Reply:
I changed the requestContext's correlation id to a Guid as it is the main mechanism in which the correlationId moves through the project. Some classes that use the correlation Id, such as the UiEvent class require a string for the correlation Id because its base type is Dictionary<string, string> and a Guid would incompatible since it cannot be added to the collection.